main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -981,6 +981,13 @@ object Weeder2 {
       flatMapN(pick(TreeKind.ArgumentList, tree))(visitArguments)
     }
 
+    /**
+     * This method is the same as pickArguments but for invokeMethod2. It calls visitMethodArguments instead.
+     */
+    private def pickMethodArguments(tree: Tree): Validation[List[Expr], CompilationMessage] = {
+      flatMapN(pick(TreeKind.ArgumentList, tree))(visitMethodArguments)
```

Comment:
We might need a new `TreeKind` here.
@herluf-ba what do you think?

Method arguments are different from normal arguments, since we do not allow them to be named (e.g., no calling `o#foo(x = 123)`)

Reply:
I think it's fine that they share the `TreeKind` so long as `visitMethodArguments` checks that no argument is named.